hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -3168,11 +3202,18 @@ func reconcileCAPIProviderDeployment(deployment *appsv1.Deployment, capiProvider
 	// Enforce ServiceAccount.
 	deployment.Spec.Template.Spec.ServiceAccountName = sa.Name
 
+	defaultSecurityContext := setDefaultSecurityContext
+
+	// For ARO HCP, the MI sidecar containers need privileged permissions to run
+	if os.Getenv("MANAGED_SERVICE") == hyperv1.AroHCP {
```

Comment:
`if os.Getenv("MANAGED_SERVICE") == hyperv1.AroHCP`
this should be a helper function, `IsAro()` or smth